openedx/core/djangoapps/external_user_ids/tests/test_batch_generate_id.py

```diff
@@ -0,0 +1,64 @@
+"""
+Test batch_get_or_create in ExternalId model
+"""
+
+from django.test import TestCase
+from common.djangoapps.student.tests.factories import UserFactory
+from openedx.core.djangoapps.external_user_ids.models import ExternalId
+from openedx.core.djangoapps.external_user_ids.tests.factories import ExternalIDTypeFactory
+
+
+class TestBatchGenerateExternalIds(TestCase):
```

Comment:
Since this is a batch operation, we should ideally add some query count test here, to make sure `more users != more queries`.

This can be done with a simple assertion in the `test_batch_get_or_create_user_ids` (see [`assertNumQueries`](https://docs.djangoproject.com/en/dev/topics/testing/tools/#django.test.TransactionTestCase.assertNumQueries), might require a change to `TransactionTestCase` instead of `TestCase`).

Reply:
Tests with the expected number of queries added. Thanks for the tip 👍🏽